lib/proxy/log.rb

```diff
@@ -12,79 +9,52 @@ def logger
   end
 
   class LoggerFactory
-    class LogFormatter < ::Logger::Formatter
-      Format = "%s, [%s%s] %5s -- %s: %s\n".freeze
-
-      def call(severity, time, progname, msg)
-        Format % [severity[0..0], format_datetime(time), request_id[0..7], severity, progname, msg2str(msg)]
-      end
-
-      def request_id
-        Thread.current.thread_variable_get(:request_id).to_s
-      end
-    end
-
     begin
       require 'syslog/logger'
-      ::Syslog::Logger.class_eval { alias_method :write, :info }
-
-      class SyslogFormatter < Syslog::Logger::Formatter
-        Format = "<%s> %s\n".freeze
-
-        def call(severity, time, progname, msg)
-          request_id.empty? ? clean(msg) : Format % [request_id[0..7], clean(msg)]
-        end
-
-        def request_id
-          Thread.current.thread_variable_get(:request_id).to_s
-        end
-      end
-
       @syslog_available = true
     rescue LoadError
-      # ignore, syslog isn't available on this platform
       @syslog_available = false
     end
 
     def self.logger
+      logger_name = 'foreman-proxy'
+      layout = Logging::Layouts.pattern(pattern: ::Proxy::SETTINGS.file_logging_pattern + "\n")
+      notime_layout = Logging::Layouts.pattern(pattern: ::Proxy::SETTINGS.system_logging_pattern + "\n")
+      logger = Logging.logger.root
       if log_file.casecmp('STDOUT').zero?
         if SETTINGS.daemon
-          puts "Settings log_file=STDOUT and daemon=true are incompatible, exiting..."
+          puts "Settings log_file=STDOUT and daemon=true cannot be used together"
           exit(1)
         end
-        logger = stdout_logger
+        logger.add_appenders(Logging.appenders.stdout(logger_name, layout: layout))
       elsif log_file.casecmp('SYSLOG').zero?
         unless syslog_available?
-          puts "'SYSLOG' logger is not supported on this platform, please use STDOUT or a file-based logger. Exiting..."
+          puts "Syslog is not supported on this platform, use STDOUT or a file"
           exit(1)
         end
-        logger = syslog_logger
+        logger.add_appenders(Logging.appenders.syslog(
+          logger_name, layout: notime_layout, facility: ::Syslog::Constants::LOG_LOCAL5))
+      elsif log_file.casecmp('JOURNAL').zero? || log_file.casecmp('JOURNALD').zero?
+        begin
+          logger.add_appenders(Logging.appenders.journald(
+            logger_name, logger_name: :proxy_logger, layout: notime_layout, facility: ::Syslog::Constants::LOG_LOCAL5))
+        rescue NoMethodError
+          logger.add_appenders(Logging.appenders.stdout(logger_name, layout: layout))
+          logger.warn "Journald is not available on this platform. Falling back to STDOUT."
+        end
       else
-        logger = default_logger(log_file)
+        begin
+          keep = ::Proxy::SETTINGS.file_rolling_keep
+          size = 1024 * 1024 * ::Proxy::SETTINGS.file_rolling_size
```

Comment:
I still think that `1024 * 1024` should be some sort of constant.
I have two reasons for my way of thinking:

1. It is easier to figure out that magic result -> more readable imo
2. If needed, it is easier to change at a single location, even if at the moment it's on a single location.

I think that performance wise it is not that much of a difference if any.

Reply:
Well I feel the opposite, it is standard practice to multiply 1024 for KiB, MiB etc. And changing that would mean an intrusive change for users (regressions), we need to agree on something (I propose MiB) and they stick with it.

Can you show a piece of code how you would like it?

Reply:
Usually I'm in favor of constants to describe what's going on, but I agree with @lzap that 1024 has been a well established pattern.

Reply:
The code I had in mind was something like this:

```ruby
BASE_LOG_SIZE = 1024 * 1024 # 1 MiB
```

It's easier to figure this out (the usage) in my opinion. And if there is a bug (just for the sake of an argument), it is easier to figure out what was the expected result, or the type of usage.

Because `1024 * 1024` for me at least when I see such calculations, feels like "magic" at first glance, even with such code that is very readable and easy to understand.

But it's just how I learned programming, that created OCD on my side ;)
It's not something that is mandatory, but nice to have imo.

Reply:
I can do this, won't hurt.

Reply:
@ik5 I get that, but now I don't know that the size is in MBs unless I look at the constant. The use of a constant is to increase readability, not decrease it.

Reply:
@ekohl, that's true for any value that is not inline.
But let's say that MiB is not the right unit, but KiB is, in this specific location, it will be easy to figure this out, but add additional place that needs the same value, and then you are starting a possible bug,that you might forgot another place, you might add constantly comment that it must be the same as the log rotation size etc... so at the end a simple constant can solve this.

But it doesn't really matter, it's just my own OCD that I created myself after long period of pain from such things, and unlike a preparation for a might be feature, it is something a bit different imo.